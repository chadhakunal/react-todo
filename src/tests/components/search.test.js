import { shallow, mount } from "enzyme";
import Search from "components/search";

const clickFn = jest.fn();

describe('Search', () => {
  it('should render correctly with no props', () => {
    const component = shallow(< Search />);
    expect(component).toMatchSnapshot();
  });

  it('should call onSearch prop when text input changes', () => {
    const component = mount(< Search onSearch={clickFn} />);
    const searchInput = component.find("input#search-input");
    searchInput.instance().value = "Hello"
    searchInput.simulate('change');
    
    expect(clickFn).toHaveBeenCalledWith("Hello")
    component.unmount();
  });
})