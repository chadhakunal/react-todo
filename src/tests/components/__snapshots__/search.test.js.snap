// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly with no props 1`] = `
<Fragment>
  <Memo(ForwardRef(SearchIcon))
    fontSize="medium"
  />
  <InputGroup
    className="w-90 search-parent"
    tag="div"
  >
    <Input
      className="search"
      id="search-input"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </InputGroup>
</Fragment>
`;
